synapse/appservice/__init__.py

```diff
@@ -35,30 +35,28 @@ def __init__(self, service, id, events):
         self.id = id
         self.events = events
 
-    def send(self, as_api):
+    async def send(self, as_api) -> bool:
```

Comment:
I'm slightly in two minds about filling out partial type hints in a function, in terms of making it less obvious which functions are typed or not. I could go either way though